test/core.js

```diff
@@ -47,6 +47,49 @@ test( "jQuery(html) loose rules", function() {
 	}
 });
 
+test( "selector state", function() {
+	expect( 18 );
+
+	var test;
+
+	test = jQuery( undefined );
+	equal( test.selector, "", "Empty jQuery Selector" );
+	equal( test.context, undefined, "Empty jQuery Context" );
+
+	test = jQuery( document );
+	equal( test.selector, "", "Document Selector" );
```

Comment:
I ran the jquery-migrate unit tests against the core PR built locally, by using the "dev" option in the jquery-migrate unit test dropdown. It seemed to pass so we should be good. We should actually land this PR first I'd say, to be sure that it works correctly with older core versions on all supported browsers.